pkg/kclient/deployments.go

```diff
@@ -18,6 +22,76 @@ func (c *Client) GetDeploymentByName(name string) (*appsv1.Deployment, error) {
 	return deployment, err
 }
 
+// getDeploymentCondition returns the condition with the provided type
+// from https://github.com/kubernetes/kubectl/blob/8bc20f428d7d5aed031de5fa160081de7b5af2b0/pkg/util/deployment/deployment.go#L58
+func getDeploymentCondition(status appsv1.DeploymentStatus, condType appsv1.DeploymentConditionType) *appsv1.DeploymentCondition {
+	for i := range status.Conditions {
+		c := status.Conditions[i]
+		if c.Type == condType {
+			return &c
+		}
+	}
+	return nil
+}
+
+// WaitForDeploymentRollout waits for deployment to finish rollout. Returns the state of the deployment after rollout.
+func (c *Client) WaitForDeploymentRollout(deploymentName string) (*appsv1.Deployment, error) {
```

Comment:
can we get a unit test for this function?

Reply:
> I can see 5 legitimate failures in the CI related to devfile @kadel .

ouch, messed something up after rebasing with latest master :-/

Reply:
Same question